sdk/resourcemanager/azure-resourcemanager-resources/src/main/java/com/azure/resourcemanager/resources/fluentcore/utils/HttpPipelineProvider.java

```diff
@@ -73,7 +72,6 @@ public static HttpPipeline buildHttpPipeline(
         if (credential != null) {
             policies.add(new AuthenticationPolicy(credential, profile.getEnvironment(), scopes));
         }
-        policies.add(new ProviderRegistrationPolicy(credential, profile));
```

Comment:
Is it possible to still keep it (automatically set the provider, after auth done)?

It still provide a function that user sometimes needs (hence we got report on the bug).

Reply:
If we want, we could set it inside `ResourceManager` constructor. If so, the `AuthorizationManager` will not has such function by default.

Reply:
Done, without `AuthorizationManager`

Reply:
Great. Remember to test it a bit (with some uncommon provider).

Reply:
Also added a check for null providers.